app/services/purchases_handler/purchases_handler.rb

```diff
@@ -0,0 +1,73 @@
+module PurchasesHandler
+  class PurchasesHandler < ApplicationService
+
+    def initialize(purchase_params)
+      @purchase_params = purchase_params
+    end
+
+    def call
+      purchase_handler
+    end
+
+    private
+
+    def purchase_handler
+      if @purchase_params[:gateway] == 'paypal'
+        cart = CartFinder.call(@purchase_params)
```

Comment:
Como o cart ta sendo utilizado em ambas condicionais, de uma lida nos callbacks do controller. o before_action é muito usado para atribuir variáveis:
http://jgaskins.org/blog/2014/08/25/better-alternative-to-rails-before_action-to-load-objects

Reply:
acabei não utilizando pois removi a repetição